### PR TITLE
feat(backup test): add storage class name from backupvolume e2e skeleton

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: [longhorn/longhorn#3506](https://github.com/longhorn/longhorn/issues/3506)

### CODE REVIEW

File: manager/integration/tests/test_basic.py [hunk 1]
```diff
@@ -1775,6 +1775,63 @@ def test_volume_multinode(client, volume_name):  # NOQA
...
+    Test the storageClasName of the restored volume's PV/PVC 
+    should be from the backup volume
+
+    Given
+    - Create a new StorageClass
+      ```
+      kind: StorageClass
+      apiVersion: storage.k8s.io/v1
+      metadata:
+        name: longhorn-backing-image-sc
```
Comment (khushboo-rancher) [inline manager/integration/tests/test_basic.py:1790]: We can change the name related to the test.

File: manager/integration/tests/test_basic.py [hunk 2]
```diff
@@ -1775,6 +1775,63 @@ def test_volume_multinode(client, volume_name):  # NOQA
...
+      provisioner: driver.longhorn.io
+      allowVolumeExpansion: true
+      reclaimPolicy: Delete
+      volumeBindingMode: Immediate
+      parameters:
+        numberOfReplicas: "2"
+        staleReplicaTimeout: "2880"
+        backingImage: "bi-download"
+        backingImageDataSourceType: "download"
+        backingImageDataSourceParameters: '{"url": "https://longhorn-backing-image.s3-us-west-1.amazonaws.com/parrot.raw"}' # NOQA
```
Comment (khushboo-rancher) [inline manager/integration/tests/test_basic.py:1800]: Are these required for this test?

File: manager/integration/tests/test_basic.py [hunk 3]
```diff
@@ -1775,6 +1775,63 @@ def test_volume_multinode(client, volume_name):  # NOQA
...
+      ```
+    - Create a PVC to use this SC
+      ```
+      apiVersion: v1
+      kind: PersistentVolumeClaim
+      metadata:
+        name: longhorn-backing-image-pvc
```
Comment (khushboo-rancher) [inline manager/integration/tests/test_basic.py:1807]: Name can be changed to something related to the test

File: manager/integration/tests/test_basic.py [hunk 4]
```diff
@@ -1775,6 +1775,63 @@ def test_volume_multinode(client, volume_name):  # NOQA
...
+    Then
+    - the backupvolume's status.storageClassName should be longhorn-backing-image-sc
+
+    When
+    - Restore the backup to a new volume
+    - Create PV/PVC from the new volume with create new PVC option
+
+    Then
+    - The new PVC's storageClassName should still be longhorn-backing-image-sc
+    """
```
Comment (khushboo-rancher) [inline manager/integration/tests/test_basic.py:1831]: Can you please add a step to verify the data

File: manager/integration/tests/test_basic.py [hunk 5]
```diff
@@ -1775,6 +1775,63 @@ def test_volume_multinode(client, volume_name):  # NOQA
...
+        name: longhorn-backing-image-pvc
+      spec:
+        accessModes:
+          - ReadWriteOnce
+        storageClassName: longhorn-backing-image-sc
+        resources:
+          requests:
+            storage: 5Gi
+      ```
+    - A BackingImage `bi-download` and a Volume will be created
```
Comment (khushboo-rancher) [inline manager/integration/tests/test_basic.py:1816]: We can avoid having additional step of creating backing image, WDYT?

**Review by khushboo-rancher — CHANGES_REQUESTED** [review]
Can we add more steps in the test to make `status.storageClassName` empty in backup and on restoring it, the storage class should pick the storage class from the default in setting?

File: manager/integration/tests/test_basic.py (hunk 2 above)
Comment (ChanYiLin) [reply manager/integration/tests/test_basic.py:1800]: You are right, we don't need backing image involved.

File: manager/integration/tests/test_basic.py (hunk 5 above)
Comment (ChanYiLin) [reply manager/integration/tests/test_basic.py:1816]: Yes I agree. Thanks

**Comment by ChanYiLin** [comment]
Hi @khushboo-rancher sorry for the late reply
for storageClassName empty case, this can only happen in upgrade after v1.5.0
for example, having a backup in v1.4.x and then upgrade to v1.5.0 then restore the volume and create the PVC

**Review by khushboo-rancher — APPROVED** [review]

**Comment by khushboo-rancher** [comment]
@mergify backport v1.5.x

**Pull request merged**